### PR TITLE
feat(auth): add pagination support to Auth user list

### LINKED ISSUE #298: Add Pagination Support to Auth Page User List
**Is your feature request related to a problem? Please describe.**

The Database page already supports pagination, but the Auth page currently displays only the first 100 users. As a result, users beyond this limit cannot be viewed or accessed from the Auth page, creating an inconsistent experience compared to the Database page.

**Describe the solution you'd like**

Add pagination support to the Auth page user list, similar to the existing implementation on the Database page. This would allow users to navigate through all available user records instead of being restricted to the first 100 entries.

**Describe alternatives you've considered**

One alternative would be increasing the number of users returned in a single request. However, implementing pagination would be more scalable, performant, and consistent with the behavior already present on the Database page.

**Additional context**

This issue was suggested by a maintainer:

> We have pagination on database page.
>
> But on auth page, it only shows the first 100 users and limits access to the next users.
>
> So we can add pagination to the auth page as well.

### Proposed Approach

* Inspect the current Auth page implementation and data-fetching logic.
* Verify whether backend pagination support already exists for the relevant endpoint.
* Reuse the existing pagination pattern from the Database page where applicable.
* Implement pagination controls on the Auth page.
* Test navigation across multiple pages and validate behavior with larger datasets.

### ETA

Tomorrow morning.

### Suggested Labels

* gssoc:approved
* type:feature
* level:beginner

### PR DESCRIPTION
**Pull request by udaycodespace**
## 🚀 Pull Request Description

Fixes #298

Implemented pagination support for the Auth page user list to make it consistent with the existing Database page pagination behavior.

### Changes Made

* Added pagination state management (`page`, `limit`, `totalRecords`) in `Auth.jsx`
* Updated Auth user fetch requests to use paginated API calls
* Added reusable Pagination component below the user table
* Added page-size selection support
* Refetches data when page or page size changes
* Preserved pagination-aware refresh after user creation

This allows users to navigate through all available auth users instead of being restricted to the first set of records.

## 🛠️ Type of Change

* [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
* [x] ✨ New feature (non-breaking change which adds functionality)
* [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] 📝 Documentation update
* [x] 🎨 UI/UX improvement (Frontend only)
* [ ] ⚙️ Refactor / Chore

### Backend Verification:

* [ ] I have run `npm test` in the `backend/` directory and all tests passed.
* [x] I have verified the API endpoints using Browser Console.
* [ ] New unit tests have been added (if applicable).

### Frontend Verification:

* [ ] I have run `npm run lint` in the `frontend/` directory.
* [ ] Verified the UI changes on different screen sizes (Responsive).
* [x] Checked for any console errors in the browser dev tools.

## 📸 Screenshots / Recordings (Pending)

⚠️ **Validation and screenshots are currently in progress.**

The current test project does not contain enough Auth users to demonstrate multi-page navigation visually.

Current verification completed:

* Confirmed the backend endpoint accepts `page` and `limit` parameters.
* Confirmed successful paginated API responses from the Auth users endpoint.
* Confirmed pagination controls render correctly on the Auth page.
* Checked browser console for errors after implementation.

Console verification output:

```text
{
  success: true,
  data: {
    items: [],
    total: 0
  },
  message: "Data fetched successfully."
}
```

I am currently preparing additional test data to fully validate page navigation behavior and will update this PR with screenshots and/or recordings once testing is completed.

## ✅ Checklist

* [x] My code follows the code style of this project.
* [x] I have performed a self-review of my code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [x] My changes generate no new warnings or errors.
* [ ] I have updated the documentation (README/Docs) accordingly.

---

Built with ❤️ for **urBackend**.

## Summary by CodeRabbit

* **New Features**
  * Client-side pagination on the Auth user management page with page/limit controls and total count.
  * New admin API endpoints to list paginated users and delete an admin user; admin user retrieval/update routes included.
  * Creates now refresh the current paginated view; changing items-per-page resets to page 1.

* **Bug Fixes**
  * Deletions update the list and total count and move to the previous page if current page becomes empty.
  * Prevents stale/overwritten results during rapid pagination changes.

### CODE REVIEW

**Comment by udaycodespace** [comment]
@yash-pouranik 

Quick update: PR is up, implementation is completed. I'll do final testing and upload screenshots/recordings tonight.

* **Problem:** Auth page only showed the first set of users and had no pagination, unlike the Database page.
* **Solution:** Reused the existing pagination pattern/component from the Database page and added paginated user fetching in Auth.
* **Status:** Code changes are complete, final validation with larger test data and screenshots is pending.

**Comment by yash-pouranik** [comment]
Some coderabbitai 
comments
@udaycodespace

**Comment by udaycodespace** [comment]
> Some coderabbitai comments @udaycodespace

- Thanks for the review. 
- I'll address the CodeRabbit suggestions, complete final validation, and update the PR with screenshots/testing results tonight!

**Comment by udaycodespace** [comment]
@yash-pouranik ,

Addressed all CodeRabbit feedback:

* Added request-id guards to prevent stale async pagination responses from overwriting newer state.
* Synced totalRecords after user deletion.
* Updated delete flow to use functional state updates and safely roll back pagination when the current page becomes empty.

Final pagination validation and screenshots are in progress.

**Comment by yash-pouranik** [comment]
I can see that only frontend is changed
does /api/userauth/*
provides pagination??
@udaycodespace

**Comment by udaycodespace** [comment]
> I can see that only frontend is changed does /api/userauth/* provides pagination?? @udaycodespace

Yes, only frontend changes were made.

Pagination is handled by the existing collection data endpoint (`/api/projects/:projectId/collections/users/data?page=&limit=`), not by `/api/userAuth/*`. The frontend now consumes that existing paginated API.

**Comment by udaycodespace** [comment]
Final validation completed.

* Created multiple Auth users and verified pagination across multiple pages.
* Confirmed page navigation (Page 1 ↔ Page 2) works correctly.
* Verified rows-per-page selection updates the dataset as expected.
* Confirmed create/delete flows keep pagination state synchronized.
* No console errors observed during testing.

Screenshots attached below.

image

image

**Comment by yash-pouranik** [comment]
2 months back when we integrated auth as a service
we created new api endpoints as
/api/userauth/*
that will be responsible for all the api calls in auth as a service
bcoz of one major reason -
/api/projects/:projectId/collections/users/data?page=&limit=
doesnt remove password like feilds from the user table and sent to the client side directly.
and 1 or 2 more reason that I was forgetting

let me know ur views @coderabbitai

**Comment by yash-pouranik** [comment]
yh all routes should go through - /admin
all listed
can u please do this?
@udaycodespace

**Comment by udaycodespace** [comment]
> yh all routes should go through - /admin all listed can u please do this? @udaycodespace

@yash-pouranik

Done. I moved the Auth user list/delete flow to dedicated `/admin/users` endpoints instead of the generic collection routes.

Also added paginated admin user APIs on the backend with password exclusion, and updated the frontend to use those endpoints consistently.

Tested locally by changing page sizes (10/20/50/100), verifying paginated refetches, and confirming list/delete requests hit `/admin/users` routes only.

**Comment by udaycodespace** [comment]
@yash-pouranik ,

- All requested changes have been addressed. 
- Auth user list/delete now use dedicated /admin/users endpoints, pagination is fully functional, passwords are excluded from responses, and all CI/CodeRabbit checks are passing. 
- Thanks for the review.

**Comment by yash-pouranik** [comment]
yh this looks more prod ready
let me check

**Comment by udaycodespace** [comment]
> yh this looks more prod ready let me check

Thanks! The implementation evolved a bit during review as well.

Besides the Auth page pagination, I also moved the user list/delete flows to dedicated `/admin/users` endpoints, added paginated backend APIs with password exclusion, added route tests, and addressed all CodeRabbit feedback around state consistency and edge cases.

Please review whenever you get a chance, and I'm happy to make any additional changes if needed.

File: apps/dashboard-api/src/controllers/userAuth.controller.js [hunk 1]
```diff
@@ -238,6 +238,64 @@ module.exports.createAdminUser = async (req, res) => {
...
+        const [items, total] = await Promise.all([
+            Model.find({}, { password: 0 }).sort({ createdAt: -1 }).skip(skip).limit(limit).lean(),
+            Model.countDocuments()
+        ]);
+
+        res.json({
+            success: true,
+            data: { items, total, page, limit },
+            message: ""
+        });
```
Comment (yash-pouranik) [inline apps/dashboard-api/src/controllers/userAuth.controller.js]: Everything looks perfect.
but I think we also have success response class.
DOnt remember. 
I think @Ayush4958 has created in earlier PR maybe?
pleae check if exist
then use it
else I will merge.

**Review by yash-pouranik — CHANGES_REQUESTED** [review]

**Comment by yash-pouranik** [comment]
@udaycodespace

**Comment by udaycodespace** [comment]
> @udaycodespace

Give me 10–15 minutes. I'll check if the success response class already exists in the project and update the PR accordingly. I'll get back to you shortly.

**Comment by Ayush4958** [comment]
> Give me 10–15 minutes. I'll check if the success response class already exists in the project and update the PR accordingly. I'll get back to you shortly.

It exist on same place where ApiError utility exist but utility name is ApiResponse

**Comment by udaycodespace** [comment]
> @udaycodespace

@yash-pouranik

Checked the codebase and confirmed that a shared `ApiResponse` utility already exists in `@urbackend/common` (same location as `ApiError`).

I've updated the admin user endpoints to use it:

* `listAdminUsers` now uses `new ApiResponse(...).send(res)`
* `deleteAdminUser` now uses `new ApiResponse(...).send(res)`

Commit: `a727453`

Thanks for pointing it out.

**Comment by udaycodespace** [comment]
> > Give me 10–15 minutes. I'll check if the success response class already exists in the project and update the PR accordingly. I'll get back to you shortly.
> 
> It exist on same place where ApiError utility exist but utility name is ApiResponse

@Ayush4958

Thanks for the quick pointer.

I checked the utility and updated the admin user endpoints to use the shared `ApiResponse` helper instead of returning the success payload directly.

Applied in commit `a727453`.

**Comment by yash-pouranik** [comment]
yh Thank you.
everything looks perf to me.

**Review by yash-pouranik — APPROVED** [review]
